sandbox/modules/identity/src/main/java/org/opensearch/identity/configuration/DynamicConfigFactory.java

```diff
@@ -95,33 +103,53 @@ public void unregisterDCFListener(Object listener) {
 
     private static class InternalUsersModelV1 extends InternalUsersModel {
 
-        private final SecurityDynamicConfiguration<User> internalUserSecurityDynamicConfiguration;
+        private final SecurityDynamicConfiguration<User> internalUserSecurityDC;
 
-        public InternalUsersModelV1(SecurityDynamicConfiguration<User> internalUserSecurityDynamicConfiguration) {
+        public InternalUsersModelV1(SecurityDynamicConfiguration<User> internalUserSecurityDC) {
             super();
-            this.internalUserSecurityDynamicConfiguration = internalUserSecurityDynamicConfiguration;
+            this.internalUserSecurityDC = internalUserSecurityDC;
         }
 
         @Override
         public User getUser(String username) {
-            return internalUserSecurityDynamicConfiguration.getCEntry(username);
+            return internalUserSecurityDC.getCEntry(username);
         }
 
         @Override
         public boolean exists(String username) {
-            return internalUserSecurityDynamicConfiguration.exists(username);
+            return internalUserSecurityDC.exists(username);
         }
 
         @Override
         public Map<String, String> getAttributes(String username) {
-            User tmp = internalUserSecurityDynamicConfiguration.getCEntry(username);
+            User tmp = internalUserSecurityDC.getCEntry(username);
             return tmp == null ? null : tmp.getAttributes();
         }
 
         @Override
         public String getHash(String username) {
-            User tmp = internalUserSecurityDynamicConfiguration.getCEntry(username);
+            User tmp = internalUserSecurityDC.getCEntry(username);
             return tmp == null ? null : tmp.getHash();
         }
     }
+
+    private static class ExtensionsSecurityConfigModelV1 extends ExtensionSecurityConfigModel {
```

Comment:
Do you mind explaining the purpose of having the versioned extension of the model? I know we have some versioned InternalUserModels but I always assumed that was from the breaking change switch and added after the fact. Is this just common practice or are you trying to prepare for other extensions of the configuration model?

Reply:
For APIs its good practice, but for here it is not necessary for the time being. I will remove the version from here.